homeassistant/components/anglian_water/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Implement a config flow for anglian_water."""
+
+from __future__ import annotations
+
+from pyanglianwater import API
+from pyanglianwater.exceptions import (
+    InvalidPasswordError,
+    InvalidUsernameError,
+    ServiceUnavailableError,
+    UnknownEndpointError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import selector
+
+from .const import CONF_DEVICE_ID, DOMAIN
+
+
+class AnglianWaterConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for anglian_water."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self,
+        user_input: dict | None = None,
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        _errors = {}
+        if user_input is not None:
+            try:
+                auth = await API.create_via_login(
+                    email=user_input[CONF_USERNAME],
+                    password=user_input[CONF_PASSWORD],
+                )
+            except InvalidUsernameError:
+                _errors["base"] = "invalid_auth"
+            except InvalidPasswordError:
+                _errors["base"] = "invalid_auth"
+            except ServiceUnavailableError:
+                _errors["base"] = "maintenance"
+            except UnknownEndpointError:
+                _errors["base"] = "cannot_connect"
+            else:
+                user_input[CONF_DEVICE_ID] = auth.device_id
```

Comment:
So you say this id is unique but only per instance?

Do we have something more unique we can fetch from the service? Like a user id?

Reply:
There is a user id, but not in the right format. The generation of this is actually performed here and is completely random so doesn't even relate to the HA instance https://github.com/pantherale0/pyanglianwater/blob/d7a9a8a6ec186a584ce8371ba32924d86faf19a4/pyanglianwater/api.py#L53

Reply:
But I mean, can we find something in their API that is unique for this account? Then we can set that as config entry unique id and prevent the same entry to be added twice, because that would mess up a lot of things

Reply:
oh right, sorry, the email is unique for an account so we can use that, was getting confused because device_id is only here for auth with the API, it has  no relation to the account and usually would identify the mobile hardware connecting.

Reply:
Can someone change their email adress via the app/service?